src/pyhf/typing.py

```diff
@@ -0,0 +1,137 @@
+import os
+import sys
+
+from typing import TYPE_CHECKING, Union, Sequence, MutableSequence
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict, Literal
+else:
+    from typing_extensions import TypedDict, Literal
```

Comment:
This is just a suggestion to avoid more import movement noise if we end up eventually turning on `isort` in `pre-commit`. Feel free to ignore though.

```suggestion
from typing import TYPE_CHECKING, MutableSequence, Sequence, Union

if sys.version_info >= (3, 8):
    from typing import Literal, TypedDict
else:
    from typing_extensions import Literal, TypedDict
```